src/sentry/flags/endpoints/hooks.py

```diff
@@ -23,52 +30,67 @@
 writing logs or leaked DSNs generating invalid log entries. An organization token is
 secret and reasonably restricted and so makes sense for this use case where we have
 inter-provider communication.
-
-This endpoint allows writes if any write-level "org" permission was provided.
 """
 
-
-class OrganizationFlagHookPermission(OrganizationPermission):
-    scope_map = {
-        "POST": ["org:ci"],
-    }
+logger = logging.getLogger()
 
 
 @region_silo_endpoint
 class OrganizationFlagsHooksEndpoint(Endpoint):
-    authentication_classes = (OrgAuthTokenAuthentication,)
+    authentication_classes = ()
     owner = ApiOwner.REPLAY
-    permission_classes = (OrganizationFlagHookPermission,)
+    permission_classes = ()
     publish_status = {
         "POST": ApiPublishStatus.PRIVATE,
     }
 
-    def convert_args(
-        self,
-        request: Request,
-        organization_id_or_slug: int | str,
-        *args,
-        **kwargs,
-    ):
+    def convert_args(self, request: Request, token: str, *args, **kwargs):
+        organization_id = get_org_id_from_token(token)
+        if not organization_id:
+            raise AuthenticationFailed("Invalid token specified.")
+
         try:
-            if isinstance(organization_id_or_slug, int):
-                organization = Organization.objects.get_from_cache(id=organization_id_or_slug)
-            else:
-                organization = Organization.objects.get_from_cache(slug=organization_id_or_slug)
+            organization = Organization.objects.get(id=organization_id)
         except Organization.DoesNotExist:
+            logger.exception("Flags Webhook: lookup failed for organization `%d`", organization_id)
             raise ResourceDoesNotExist
 
-        self.check_object_permissions(request, organization)
-        bind_organization_context(organization)
-
         kwargs["organization"] = organization
         return args, kwargs
 
     def post(self, request: Request, organization: Organization, provider: str) -> Response:
+        if not features.has(
+            "organizations:feature-flag-audit-log", organization, actor=request.user
+        ):
+            return Response("Not enabled.", status=404)
+
         try:
             write(handle_provider_event(provider, request.data, organization.id))
             return Response(status=200)
         except InvalidProvider:
             raise ResourceDoesNotExist
         except DeserializationError as exc:
-            return Response(exc.errors, status=400)
+            sentry_sdk.capture_exception()
+            return Response(exc.errors, status=200)
```

Comment:
If the response has errors it probably shouldn't also be a 200.

Reply:
We don't want hook providers to make more than one request if its a deterministic failure.